python_modules/dagster/dagster/core/definitions/decorators/solid_decorator.py

```diff
@@ -373,6 +374,20 @@ def resolve_checked_solid_fn_inputs(
                 used_inputs.add(param.name)
 
     undeclared_inputs = explicit_names - used_inputs
+
+    if decorator_name == "@graph":
```

Comment:
i would leave a little comment here explaining the context for why we raise a special error here before the other one below